src/pkg/immutabletag/controller.go

```diff
@@ -0,0 +1,70 @@
+package immutabletag
+
+import (
+	"github.com/goharbor/harbor/src/pkg/immutabletag/model"
+)
+
+// APIController to handle the requests related with immutabletag
+type APIController interface {
+	// GetImmutableRule ...
+	GetImmutableRule(id int64) (*model.Metadata, error)
+
+	// CreateImmutableRule ...
+	CreateImmutableRule(m *model.Metadata) (int64, error)
+
+	// DeleteImmutableRule ...
+	DeleteImmutableRule(id int64) error
+
+	// UpdateImmutableRule ...
+	UpdateImmutableRule(pid int64, m *model.Metadata) error
+
+	// ListImmutableRules ...
+	ListImmutableRules(pid int64) ([]model.Metadata, error)
+}
+
+// DefaultAPIController ...
+type DefaultAPIController struct {
+	manager Manager
```

Comment:
The controller just delegate the request for manager. do we need controller for all modules?